tests/test_fee_burner.py

```diff
@@ -0,0 +1,377 @@
+    # given: a user adds adds allowance on OMG contract and user has some initial tokens
+    omg_token.functions.approve(fee_burner.address, HUGE_AMOUNT).transact({'from': non_operator})
+    fee_burner.functions.receive().transact({'value': 100000})
+
+    user_initial_balance = w3.eth.getBalance(non_operator)
+
+    # when: the user sends an exchange demand OMG for other token at rate 1,1 (initial rate)
+    fee_burner.functions.exchange(1, 1, 100000, 100000).transact({'from': non_operator})
+
+    # then: user has received token and OMGs have been burnt
+    assert omg_token.functions.balanceOf(DEAD_ADDRESS).call() == 100000
+    assert w3.eth.getBalance(non_operator) > user_initial_balance
+    assert w3.eth.getBalance(non_operator) <= user_initial_balance + 100000
+
+
+def test_exchange_when_user_offers_more_omgs_then_needed(w3, non_operator, omg_token, fee_burner, other_token):
+    # given: added support for some token and
+    fee_burner.functions.addSupportFor(other_token.address, 1, 1).transact()
+    other_token.functions.transfer(fee_burner.address, SMALL_AMOUNT).transact()
+
+    # given: a user adds adds allowance on OMG contract and user has some initial tokens
+    omg_token.functions.approve(fee_burner.address, HUGE_AMOUNT).transact({'from': non_operator})
+    user_initial_balance = other_token.functions.balanceOf(non_operator).call()
+
+    # when: the user sends offer at valid rate, but sends to much OMGs
+    fee_burner.functions.exchange(other_token.address, 1, 1, 200, 1).transact({'from': non_operator})
+
+    # then: the transaction was valid and tokens were transferred
+    assert omg_token.functions.balanceOf(DEAD_ADDRESS).call() == 200
+    assert other_token.functions.balanceOf(non_operator).call() == user_initial_balance + 1
+
+
+def test_exchange_at_invalid_rate(non_operator, omg_token, fee_burner, other_token):
+    # given: added support for some token and
+    fee_burner.functions.addSupportFor(other_token.address, 1, 1).transact()
+    other_token.functions.transfer(fee_burner.address, SMALL_AMOUNT).transact()
+
+    # given: a user adds adds allowance on OMG contract and user has some initial tokens
+    omg_token.functions.approve(fee_burner.address, SMALL_AMOUNT).transact({'from': non_operator})
+    user_initial_balance = other_token.functions.balanceOf(non_operator).call()
+
+    # when: the user sends an exchange demand OMG for other token at rate 2,1 (not current rate)
+    # then: error
+    with pytest.raises(ValueError):
+        fee_burner.functions.exchange(other_token.address, 2, 1, 2, 1).transact({'from': non_operator})
+
+    # when: the user sends an exchange demand OMG for other token at valid rate, but amounts aren't at this rate
+    # then: error
+    with pytest.raises(ValueError):
+        fee_burner.functions.exchange(other_token.address, 1, 1, 1, 2).transact({'from': non_operator})
+
+    # TODO: probably I needn't check this
```

Comment:
I think I wouldn't. I would only if you did a similar maneouver as in HonteStaking tests, where the `transacts` were wrapped and the wrappers checked all the post conditions after a transaction. HonteStaking had this only for successful transactions, but you can also have a failure-expecting wrapper, that works the similar way. Such wrapper would then check post-conditions like these.